mysql-test/suite/innodb/t/percona_sys_tablespace_encrypt.test

```diff
@@ -222,6 +254,86 @@ SELECT @@innodb_sys_tablespace_encrypt;
+
+SET @save_innodb_file_per_table = @@global.innodb_file_per_table;
+SET GLOBAL innodb_file_per_table = 0;
+
+--echo # Setting innodb_file_per_table=0 forces table to be created using system tablespace unless
+--echo # explicilty tablespace=innodb_file_per_table is mentioned
+
+--echo # Must fail as it tries to create table in system tablespace which is a shared tablespace
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TABLE t2 (a int) ENCRYPTION='Y';
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TABLE t3 (a int) ENCRYPTION='N';
+
+--echo # Must be successful
+CREATE TABLE t2 (a int) TABLESPACE = innodb_file_per_table, ENCRYPTION='Y';
+CREATE TABLE t3 (a int) TABLESPACE = innodb_file_per_table, ENCRYPTION='N';
+DROP TABLE t2,t3;
+
+CREATE TABLE t2 (a int);
+--echo # Try to move the table from encrypted sys to un-encrypted file_per_table space 
+--echo # Must fail as the source is encrypted but the destination is not
+--error ER_TARGET_TS_UNENCRYPTED
+ALTER TABLE t2 TABLESPACE = innodb_file_per_table;
+
+--echo # Try to move the table from encrypted sys to encrypted file_per_table space
+--echo # Must be successful
+ALTER TABLE t2 TABLESPACE = innodb_file_per_table, ENCRYPTION='Y';
+DROP TABLE t2;
+
+--echo # Try to move the table from encrypted sys to un-encrypted external tablespace
+--echo # Must fail as the source is encrypted but the destination is not
+CREATE TABLESPACE tb01 ADD DATAFILE 'tb01.ibd' ENGINE=INNODB;
+CREATE TABLE t2 (a int);
+--error ER_TARGET_TS_UNENCRYPTED
+ALTER TABLE t2 TABLESPACE = tb01;
+DROP TABLESPACE tb01;
+DROP TABLE t2;
+
+--echo # Try to move the table from encrypted sys to encrypted external tablespace
+--echo # Must be successful as both the source & destination are encrypted
+CREATE TABLESPACE tb01 ADD DATAFILE 'tb01.ibd' ENCRYPTION='Y' ENGINE=INNODB;
+CREATE TABLE t2 (a int);
+ALTER TABLE t2 TABLESPACE = tb01;
+DROP TABLE t2;
+DROP TABLESPACE tb01;
+ 
+SET GLOBAL innodb_file_per_table = @save_innodb_file_per_table;
+
+###################### Additional Test Scenarios #####################################
+# 3. Move table from encrypted sys tablespace to un-encrypted tablespace             #
+# 4. Move table from encrypted sys tablespace to encrypted tablespace                #
+######################################################################################
+
+--echo # 3. Move table from encrypted sys tablespace to un-encrypted tablespace
+CREATE TABLE t2 (a int) TABLESPACE = innodb_system;
+--error ER_TARGET_TS_UNENCRYPTED
+ALTER TABLE t2 TABLESPACE = innodb_file_per_table;
+DROP TABLE t2;
+
+CREATE TABLE t2 (a int) TABLESPACE = innodb_system;
+CREATE TABLESPACE tb01 ADD DATAFILE 'tb01.ibd';
+--error ER_TARGET_TS_UNENCRYPTED
+ALTER TABLE t2 TABLESPACE = tb01;
+DROP TABLE t2;
+
+--echo # 4. Move table from encrypted sys tablespace to encrypted tablespace
+CREATE TABLE t2 (a int) TABLESPACE = innodb_system;
+ALTER TABLE t2 TABLESPACE = innodb_file_per_table, ENCRYPTION='Y';
```

Comment:
already covered:

ALTER TABLE t7 TABLESPACE=`innodb_file_per_table`, ENCRYPTION='Y';
SHOW CREATE TABLE t7;

Reply:
Removed this , but retained
--echo # 4. Move table from encrypted sys tablespace to encrypted tablespace
CREATE TABLE t2 (a int) TABLESPACE = innodb_system;
CREATE TABLESPACE tb01 ADD DATAFILE 'tb01.ibd' ENCRYPTION='Y';
ALTER TABLE t2 TABLESPACE = tb01;
DROP TABLE t2; 
DROP TABLESPACE tb01;